webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -1420,34 +1403,50 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
       * Helper method for checking if an username is already registered.
       *
       * @param username the username of the user.
+      * @param current the current username of the user.
       * @return a [[Boolean]].
       */
-    private def userByUsernameExists(username: String): Future[Boolean] = {
-        for {
-            askString <- Future(queries.sparql.admin.txt.checkUserExistsByUsername(username = username).toString)
-            // _ = log.debug("userExists - query: {}", askString)
+    private def userByUsernameExists(username: Option[String], current: Option[String] = None): Future[Boolean] = {
```

Comment:
Why use an `Option` for `username`? Is there a reason to call this function if you don't have a new username?

Reply:
Because the username might or might not be defined in the request. And if I check this before calling the function, the exception is ignored.

Reply:
OK, now I understand, thanks.